server/src/main/java/io/mantisrx/server/master/domain/DataFormatAdapter.java

```diff
@@ -352,14 +356,21 @@ public static JobWorker convertMantisWorkerMetadataWriteableToMantisWorkerMetada
                 .withSlaveID(writeable.getSlaveID())
                 .withStageNum(writeable.getStageNum())
 
-                .withState(convertMantisJobStateToWorkerState(writeable.getState()))
                 .withWorkerIndex(writeable.getWorkerIndex())
                 .withWorkerNumber(writeable.getWorkerNumber())
 
                 .withJobCompletedReason(writeable.getReason())
                 .withPreferredCluster(writeable.getCluster())
-                .withLifecycleEventsPublisher(eventPublisher)
-                .build();
+                .withLifecycleEventsPublisher(eventPublisher);
+
+        if (workerPorts == null && !MantisJobState.isTerminalState(writeable.getState())) {
```

Comment:
@nickmahilani @neerajrj a last iteration:

I'm explicitly marking the worker as `FAILED` if there's no ports. When the worker gets loaded, the scheduler will notice that it's in a bad state and resubmit. With this approach, we don't have to do anymore weird Null checking before sending scheduling info.

Reply:
Worker ports should be checked only for workers in Launched/StartInitiated/Started state. Would this affect workers in all states?

Reply:
It would, but also `Accepted`. From your other suggestion, I'll change this to only use Launched/StartInitiated/Started via `MantisJobState.isRunningState`. Unit test for this to come shortly.